src/rssitem.cpp

```diff
@@ -176,13 +157,13 @@ bool RssItem::has_attribute(const std::string& attribname)
 std::string RssItem::get_attribute(const std::string& attribname)
 {
 	if (attribname == "title") {
-		return title();
+		return utils::utf8_to_locale(title());
 	} else if (attribname == "link") {
 		return link();
```

Comment:
> In Newsboat, all strings are in UTF-8

Is there any chance the string returned by  `link()` (and ofther calls below like `guid`, `guid`, etc.) also need to be converted from utf8 to the user's locale?

Reply:
i was refactoring, not looking for bugs :) but let's see:

- `link`: perhaps. It's copied verbatim from the feed, and theoretically can be in any encoding (although "proper" URLs will be all-ASCII, using percent-encoding for non-ASCII parts). I think the correct solution here would be to enforce percent-encoding in RSS/Atom parsers

- `date`: not sure. I didn't test that, but I bet `localtime` returns a correctly-encoded string already. Or maybe it's always ASCII

- `guid`: we can't really know. This is an arbitrary string copied from the feed, and we don't know its encoding. A correct feed would encode it the same as the rest of XML, but completely correct feeds are rare enough that I won't count of this.

- `unread`: no. It's one of two ASCII constants "yes" and "no"

- `enclosure_url`: same as `link`

- `enclosure_type`: no. It's one of a few pre-defined ASCII strings

- `flags`: no. It's a set of Latin letters

- `age`: no. It's a number using Arabic numerals

- `articleindex`: same as `age`

(The above assumes that all encodings are compatible with ASCII. I don't think it's true, but I never received any bug reports about this, so either these encodings don't exist, they aren't in use, or people who use them have more pressing issues with Newsboat than that).